src/Microsoft.OpenApi/Models/OpenApiCallback.cs

```diff
@@ -34,6 +34,20 @@ public class OpenApiCallback : IOpenApiSerializable, IOpenApiReferenceable, IOpe
         /// This object MAY be extended with Specification Extensions.
         /// </summary>
         public IDictionary<string, IOpenApiExtension> Extensions { get; set; } = new Dictionary<string, IOpenApiExtension>();
+        
+        /// <summary>
+        /// Initializes a copy of an <see cref="OpenApiCallback"/> object
+        /// </summary>
+        public object Clone()
+        {
+            return new OpenApiCallback
+            {
+                PathItems = PathItems,
```

Comment:
all the properties need to use the clone method before being assigned to achieve a deep copy (this only has been done for strings at the moment). Right now the code is making reference copies which could have side effects as described in an earlier comment.